### PR TITLE
Reorganize the ndm service path for implement dummy function

### PR DESCRIPTION
**Pull request by JamePeng**
SuspendDaemons , ResumeDaemons , OverrideDefaultDaemons

The NDM file move to /core/hle/service/ndm/ now!

Signed-off-by: JamePeng <jame_peng@sina.com>

### CODE REVIEW

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-14T12:02:30Z: https://github.com/citra-emu/citra/compare/0d086616d1af239b8e41b246bea3c1ef85fc907f...JamePeng:dc6d71a42239604c4e0d639bef59a48fe120c50b

File: src/core/hle/service/ndm/ndm.h [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+enum class DaemonMask :u32
+{
+    DAEMON_MASK_CEC     = (1 << 0),
+    DAEMON_MASK_BOSS    = (1 << 1),
+    DAEMON_MASK_NIM     = (1 << 2),
+    DAEMON_MASK_FRIENDS = (1 << 3)
+};
+
+enum class DaemonStatus :u32
```
Comment (lioncash) [inline src/core/hle/service/ndm/ndm.h]:
```cpp
enum class DaemonStatus : u32
{
    Busy,
    Idle,
    Suspending,
    Suspended
};
```

File: src/core/hle/service/ndm/ndm.h [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+enum class Daemon :u32
+{
+    DAEMON_CEC     = 0,
+    DAEMON_BOSS    = 1,
+    DAEMON_NIM     = 2,
+    DAEMON_FRIENDS = 3
+};
+
+enum class DaemonMask :u32
```
Comment (lioncash) [inline src/core/hle/service/ndm/ndm.h]:
```cpp
enum class DaemonMask :u32
{
    CEC     = (1 << 0),
    BOSS    = (1 << 1),
    NIM     = (1 << 2),
    FRIENDS = (1 << 3)
};
```

File: src/core/hle/service/ndm/ndm.h [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+#include "common/logging/log.h"
+
+namespace Service {
+
+class Interface;
+
+namespace NDM {
+
+enum class Daemon :u32
```
Comment (lioncash) [inline src/core/hle/service/ndm/ndm.h]:
```cpp
enum class Daemon : u32
{
    CEC,
    BOSS,
    NIM,
    FRIENDS
};
```

File: src/core/hle/service/ndm/ndm.h [hunk 4]
```diff
@@ -0,0 +1,78 @@
+// Copyright 2015 Citra Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include "common/common_types.h"
+
+#include "common/logging/log.h"
```
Comment (lioncash) [inline src/core/hle/service/ndm/ndm.h]: This include belongs in the cpp file.

File: src/core/hle/service/ndm/ndm.cpp [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+void Init() {
+    AddService(new NDM_U_Interface);
+}
+
+void Shutdown() {
+
+}
+
+}// namespace NDM
+}// namespace Service
```
Comment (lioncash) [inline src/core/hle/service/ndm/ndm.cpp]: Missing newline

File: src/core/hle/service/ndm/ndm_u.cpp [hunk 6]
```diff
@@ -29,17 +28,15 @@ const Interface::FunctionInfo FunctionTable[] = {
...
-// Interface class
-
-Interface::Interface() {
+NDM_U_Interface::NDM_U_Interface() {
     Register(FunctionTable);
 }
 
-} // namespace
+} // namespace NDM
+} // namespace Service
```
Comment (lioncash) [inline src/core/hle/service/ndm/ndm_u.cpp]: Missing newline

Comment (JamePeng) [inline src/core/hle/service/ndm/ndm_u.cpp]: @lioncash CMAKE detected the "missing file", what can I do for it?

Comment (JamePeng) [inline src/core/hle/service/ndm/ndm_u.cpp]: Oh, core have a CMAKE file!

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-14T12:28:46Z: https://github.com/citra-emu/citra/compare/0d086616d1af239b8e41b246bea3c1ef85fc907f...JamePeng:02a227a8520072873163c84796135fd1382cd6a4

File: src/core/hle/service/ndm/ndm.h [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+{
+    CEC,
+    BOSS,
+    NIM,
+    FRIENDS
+};
+
+enum class DaemonMask : u32
+{
+    CEC = (1 << 0),
```
Comment (lioncash) [inline src/core/hle/service/ndm/ndm.h]: OK, now realign the `=` please. Like the one in the example I gave you

Comment (JamePeng) [inline src/core/hle/service/ndm/ndm.h]: Ok!

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-14T12:34:32Z: https://github.com/citra-emu/citra/compare/0d086616d1af239b8e41b246bea3c1ef85fc907f...JamePeng:01cc779dca36fc90f49dff71ffa1f2b755fd188c

File: src/core/hle/service/ndm/ndm.cpp [hunk 8]
```diff
@@ -0,0 +1,48 @@
+// Copyright 2015 Citra Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#include <cinttypes>
```
Comment (lioncash) [inline src/core/hle/service/ndm/ndm.cpp]: cinttypes isn't needed. you need to include `common/common_types.h`

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-14T12:45:33Z: https://github.com/citra-emu/citra/compare/0d086616d1af239b8e41b246bea3c1ef85fc907f...JamePeng:1e7c16f9ac830ff5f16cec55d8c712715f4f387d

File: src/core/hle/service/ndm/ndm.cpp (hunk 8 above)
Comment (JamePeng) [inline src/core/hle/service/ndm/ndm.cpp]: got it!

**Comment by JamePeng** [comment]
@lioncash How about this time? I think better than last time!

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-14T14:14:56Z: https://github.com/citra-emu/citra/compare/0d086616d1af239b8e41b246bea3c1ef85fc907f...JamePeng:858a451187695be4d972ed9e5ec688b55b9cdeed

**Comment by JamePeng** [comment]
Ready for merge!

File: src/core/hle/service/ndm/ndm.cpp [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+#include "core/hle/service/ndm/ndm.h"
+#include "core/hle/service/ndm/ndm_u.h"
+
+namespace Service {
+namespace NDM {
+
+void SuspendDaemons(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+
+    LOG_WARNING(Service_NDM, "(STUBBED DaemonBitMask=0x%08X ", cmd_buff[1]);
```
Comment (Subv) [inline src/core/hle/service/ndm/ndm.cpp]: Missing ) on all the logs

Comment (JamePeng) [inline src/core/hle/service/ndm/ndm.cpp]: @Subv hmm,right (STUB)

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-26T21:28:06Z: https://github.com/citra-emu/citra/compare/2f56125f12ed2a9a18226f968b44d0e0a36bdd91...JamePeng:efc6b2d0ffb121a00bcf6d499457bf8f37dbaae2

**Comment by JamePeng** [comment]
Ready for merge!

File: src/core/hle/service/ndm/ndm.h [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+#include "common/common_types.h"
+
+namespace Service {
+
+class Interface;
+
+namespace NDM {
+
+enum class Daemon : u32
```
Comment (bunnei) [inline src/core/hle/service/ndm/ndm.h]: You don't use this struct, it doesn't really have any value to be here unused. I'd take it out.

File: src/core/hle/service/ndm/ndm.h [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+{
+    CEC,
+    BOSS,
+    NIM,
+    FRIENDS
+};
+
+enum class DaemonMask : u32
```
Comment (bunnei) [inline src/core/hle/service/ndm/ndm.h]: Ditto

File: src/core/hle/service/ndm/ndm.h [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+{
+    CEC     = (1 << 0),
+    BOSS    = (1 << 1),
+    NIM     = (1 << 2),
+    FRIENDS = (1 << 3)
+};
+
+enum class DaemonStatus : u32
```
Comment (bunnei) [inline src/core/hle/service/ndm/ndm.h]: Ditto

File: src/core/hle/service/ndm/ndm.h [hunk 13]
```diff
@@ -0,0 +1,76 @@
+// Copyright 2015 Citra Emulator Project
```
Comment (bunnei) [inline src/core/hle/service/ndm/ndm.h]: 2016

File: src/core/hle/service/ndm/ndm.cpp [hunk 14]
```diff
@@ -0,0 +1,47 @@
+// Copyright 2015 Citra Emulator Project
```
Comment (bunnei) [inline src/core/hle/service/ndm/ndm.cpp]: 2016

File: src/core/hle/service/ndm/ndm.cpp [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+#include "core/hle/service/ndm/ndm.h"
+#include "core/hle/service/ndm/ndm_u.h"
+
+namespace Service {
+namespace NDM {
+
+void SuspendDaemons(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+
+    LOG_WARNING(Service_NDM, "(STUBBED) DaemonBitMask=0x%08X ", cmd_buff[1]);
```
Comment (bunnei) [inline src/core/hle/service/ndm/ndm.cpp]: Change "DaemonBitMask" to just "bit_mask"

File: src/core/hle/service/ndm/ndm.cpp [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+
+    cmd_buff[1] = RESULT_SUCCESS.raw; // No error
+}
+
+void ResumeDaemons(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+
+    LOG_WARNING(Service_NDM, "(STUBBED) DaemonBitMask=0x%08X ", cmd_buff[1]);
```
Comment (bunnei) [inline src/core/hle/service/ndm/ndm.cpp]: Change "DaemonBitMask" to just "bit_mask"

File: src/core/hle/service/ndm/ndm.cpp [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+
+    cmd_buff[1] = RESULT_SUCCESS.raw; // No error
+}
+
+void OverrideDefaultDaemons(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+
+    LOG_WARNING(Service_NDM, "(STUBBED) DaemonBitMask=0x%08X ", cmd_buff[1]);
```
Comment (bunnei) [inline src/core/hle/service/ndm/ndm.cpp]: Change "DaemonBitMask" to just "bit_mask"

**Comment by bunnei** [comment]
LGTM

**Pull request merged**